Code/GraphMol/ROMol.cpp

```diff
@@ -667,4 +674,29 @@ unsigned int ROMol::addConformer(Conformer *conf, bool assignId) {
   return conf->getId();
 }
 
+#ifdef RDK_USE_BOOST_SERIALIZATION
+template <class Archive>
+void ROMol::save(Archive &ar, const unsigned int) const {
+  std::string pkl;
+  MolPickler::pickleMol(*this, pkl, PicklerOps::AllProps);
+  ar << pkl;
+}
+
+template <class Archive>
+void ROMol::load(Archive &ar, const unsigned int) {
+  std::string pkl;
+  ar >> pkl;
+
+  initMol();
+  numBonds = 0;
+  MolPickler::molFromPickle(pkl, *this, PicklerOps::AllProps);
```

Comment:
Do we know what happens when this fails?

Reply:
I assume it throws the same exception you'd get if you pass the same bad pickle data to `ROMol(std::string)`